dcc-portal-server/src/main/java/org/icgc/dcc/portal/server/pql/convert/FiltersConverter.java

```diff
@@ -115,28 +119,57 @@ public String convertFilters(JqlFilters filters, Type indexType) {
     }
 
     // These fields are required to create the correct nesting order. E.g. nested(gene, nested(gene.ssm ...))
-    val fieldsGrouppedByNestedPath = ArrayListMultimap.<String, JqlField> create();
+    val fieldsGroupedByNestedPath = groupFields(filters, indexType, f -> f.getOperation() != NOT);
+    val notFieldsGroupedByNestedPath = groupFields(filters, indexType, f -> f.getOperation() == NOT);
+
+    val groupedFilters = getGroupedFilters(fieldsGroupedByNestedPath, indexType, false);
+    val notFilters = getGroupedFilters(notFieldsGroupedByNestedPath, indexType, true);
+
+    return Stream.concat(groupedFilters.values().stream(), notFilters.values().stream()).collect(joining(","));
+  }
+
+  private ListMultimap<String, JqlField> groupFields(JqlFilters filters, Type indexType,
+      Predicate<JqlField> streamFilter) {
+    val groupedByNestedPath = ArrayListMultimap.<String, JqlField> create();
 
-    // filterKind is 'donor' in a filter {donor:{id:{is:'DO1'}}}
     for (val entry : filters.getKindValues().entrySet()) {
-      val fieldsByNestedPath = groupFieldsByNestedPath(entry.getKey(), entry.getValue(), indexType);
-      fieldsGrouppedByNestedPath.putAll(fieldsByNestedPath);
+      // This guy needs to be mutable because groupFieldsByNestedPath method mutates it.
+      List<JqlField> values = entry.getValue().stream()
+          .filter(streamFilter)
+          // We transform NOT operations to IS operations as we are moving the NOT higher up in the AST.
+          .map(f -> f.getOperation() == NOT ? new JqlField(f.getName(), IS, f.getValue(), f.getPrefix()) : f)
```

Comment:
I would extract method on this so you can do:

```java
val fields = entry.getValue().stream().map(this::removeNot).collect(toList())
```
Easier to visually scan and understand.